src/lua/httpc.lua

```diff
@@ -228,6 +241,56 @@ local function process_headers(headers)
     return headers
 end
 
+-- In the Extended BNF notation of RFC 1341, a Content-Type header field value
+-- is defined as follows: Content-Type := type "/" subtype *[";" parameter].
+-- See RFC 1341, https://www.w3.org/Protocols/rfc1341/4_Content-Type.html
+-- Function splits content-type header for part with type and subtype and part
+-- parameter and returns the first one.
+local function header_value_wo_parameters(header_value)
+    if header_value == nil then
+        return header_value
+    end
+    if type(header_value) ~= "string" then
+        error('header value must be a string', 2)
+    end
+    local res = string.split(header_value, ';')
+    local header_value = res[1]
+    return header_value:gsub("^%s+", ""):gsub("%s+$", "")
```

Comment:
`string.split()` has third argument, which allows to stop splitting after N iterations if don't interested in more than (N+1) parts.

There are several ways to write this code simpler:

```lua
return header_value:split(';', 1)[1]:strip()
```

```lua
return header_value:match('^(.-);'):strip()
```

What confuses me in current code:

* `res` is not a function result.
* Redefinition of the `header_value` with a value of different meaning.
* Using two regexps instead of the built-in strip function.